SignalShareExtension/ShareViewController.swift

```diff
@@ -448,7 +450,31 @@ public class ShareViewController: UINavigationController, ShareViewDelegate, SAE
         // TODO accept other data types
         // TODO whitelist attachment types
         // TODO coerce when necessary and possible
-        return promise.then { (url: URL) -> SignalAttachment in
+        return promise.then { (itemUrl: URL) -> Promise<SignalAttachment> in
+
+            let url: URL = try {
+                // iOS converts at least some video formats (e.g. com.apple.quicktime-movie) into mp4s as part of the
+                // NSItemProvider `loadItem` API.
+                // However, for some reason, AVFoundation operations such as generating a preview image and playing
+                // the url in the AVMoviePlayer fail on these converted formats until unless we first copy the media
+                // into our container. (These operations succeed when sending a non-converted mp4 (e.g. one received,
+                // saved, and resent in Signal)
+                //
+                // I don't understand why this is, and I haven't found any relevant documentation in the NSItemProvider
+                // or AVFoundation docs.
+                //
+                // I *did* verify that the size and sah256 sum of the original url matches that of the copied url.
+                // Perhaps the AVFoundation API's require some extra file system permssion we don't have in the
+                // passed through URL.
+                if MIMETypeUtil.isSupportedVideoFile(itemUrl.path) {
+                    return try SignalAttachment.copyToVideoTempDir(url: itemUrl)
```

Comment:
Isn't this backwards?  It reads as though you "copy to temp" if the URL _IS_ supported.  I thought the copy was only necessary if the URL wasn't supported, e.g. if the video has just been converted by iOS.

Reply:
It seems like we'll want to `setShouldDeleteOnDeallocation` IFF you copied to temp.

Reply:
We need to do it for basically *any* video. It's not related to the file type - I think it's related to where the file lives on disk. I agree that it's confusing as written. I'll clarify this.

Reply:
Okay, just make sure you only use `setShouldDeleteOnDeallocation` for temp files - not originals.